servicesCrud/src/services/repository.ts

```diff
@@ -0,0 +1,24 @@
+import { IService } from './interface';
+import { ServiceModel } from './model';
+
+export class ServiceRepository {
+  static create(newservice: IService): Promise<IService> {
+    return ServiceModel.create(newservice);
+  }
+  
+  static updateServiceById(newservice: Partial<IService>, id: string): Promise<IService | null> {
+    return ServiceModel.findByIdAndUpdate(id, newservice, { new: true }).exec();
+  }
+
+  static getServiceById(serviceId: string): Promise<IService | null> {
+    return ServiceModel.findById(serviceId).exec();
+  }
+  
+  static getAll(): Promise<IService[]> {
+    return ServiceModel.find({}).exec();
```

Comment:
notice here you don't need to have `orFail` because if `getAll` doesn't find anything it returns an empty array
```suggestion
  static getAll(query: Partial<IService>): Promise<IService[]> {
    return ServiceModel.find(query).exec();
```